src/oss/langchain/multi-agent.mdx

```diff
@@ -295,23 +511,1231 @@ const callSubagent1 = tool(
+  },
+  {
+    name: "record_warranty_status",
+    description: "Record warranty status and transition",
+    schema: z.object({ status: z.string() })
+  }
+);
+
+// 3. Middleware applies dynamic configuration based on currentStep
+const applyStepConfig = wrapModelCall(  // [!code highlight]
+  async (request: ModelRequest, handler) => {
+    const step = request.state.currentStep || "triage";  // [!code highlight]
+
+    // Map steps to their configurations
+    const configs = {
+      triage: {
+        prompt: "Collect warranty information...",
+        tools: [recordWarrantyStatus]
+      },
+      specialist: {
+        prompt: `Provide solutions based on warranty: ${request.state.warrantyStatus}`,
+        tools: [provideSolution, escalate]
+      }
+    };
+
+    const config = configs[step];
+    request = request.override({  // [!code highlight]
+      systemPrompt: config.prompt,  // [!code highlight]
+      tools: config.tools  // [!code highlight]
+    });
+    return handler(request);
+  }
+);
+
+// 4. Create agent with middleware
+const agent = createAgent({
+  model,
+  tools: [recordWarrantyStatus, provideSolution, escalate],
+  stateSchema: SupportState,
+  middleware: [applyStepConfig],  // [!code highlight]
+  checkpointer: new InMemorySaver()  // Persist state across turns  // [!code highlight]
+});
+```
+:::
+
+</Accordion>
+
+### Multiple agent subgraphs
+
+Multiple distinct agents exist as separate nodes in a graph. Handoff tools navigate between agent nodes using `Command.PARENT` to specify which node to execute next:
+
+:::python
+```python
+@tool
+def transfer_to_sales():
+    """Transfer to the sales agent."""
+    return Command(
+        goto="sales_agent",  # Navigate to the sales agent node
+        update={"active_agent": "sales_agent"},
+        graph=Command.PARENT  # Navigate in parent graph
+    )
+```
+:::
+:::js
+```typescript
+const transferToSales = tool(
+  async () => {
+    return new Command({
+      goto: "sales_agent",  // Navigate to the sales agent node
+      update: { activeAgent: "sales_agent" },
+      graph: Command.PARENT  // Navigate in parent graph
+    });
+  },
+  {
+    name: "transfer_to_sales",
+    description: "Transfer to the sales agent.",
+    schema: z.object({})
+  }
+);
+```
+:::
+
+<Accordion title="Complete example: Sales and support with handoffs">
+
+This example shows a multi-agent system with separate sales and support agents. Each agent is a separate graph node, and handoff tools allow agents to transfer conversations to each other.
+
+:::python
+```python
+from typing import Literal
+from langchain.agents import AgentState, create_agent
+from langchain.tools import tool
+from langgraph.graph import StateGraph, START
+from langgraph.types import Command
+
+# 1. Define state with active_agent tracker
+class MultiAgentState(AgentState):
+    active_agent: str = "sales_agent"  # Track which agent is active
+
+# 2. Create handoff tools
+@tool
+def transfer_to_sales():
```

Comment:
We should probably include an update of message history to include the AIMessage and the ToolMessage associated w/ the transfer